databricks/koalas/series.py

```diff
@@ -4168,7 +4168,7 @@ def _cum(self, func, skipna, part_cols=()):
         # This is used to cummin, cummax, cumsum, etc.
         index_columns = self._internal.index_columns
         window = Window.orderBy(
-            index_columns).partitionBy(*part_cols).rowsBetween(
+            F.monotonically_increasing_id()).partitionBy(*part_cols).rowsBetween(
```

Comment:
We can't simply use this here since the plan will be messy.

E.g.

```py
kdf = ks.DataFrame([[2.0, 1.0], [3.0, None], [1.0, 0.0]], columns=list('AB'))
>>> kdf.cummax().explain()
== Physical Plan ==
*(4) Project [__index_level_0__#0L, _we0#40 AS A#35, CASE WHEN isnull(B#2) THEN null ELSE _we1#41 END AS B#37]
+- Window [max(A#1) windowspecdefinition(_w1#38L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS _we0#40], [_w1#38L ASC NULLS FIRST]
   +- *(3) Sort [_w1#38L ASC NULLS FIRST], false, 0
      +- *(3) Project [__index_level_0__#0L, A#1, _w1#38L, B#2, _we1#41]
         +- Window [max(B#2) windowspecdefinition(_w3#39L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS _we1#41], [_w3#39L ASC NULLS FIRST]
            +- *(2) Sort [_w3#39L ASC NULLS FIRST], false, 0
               +- Exchange SinglePartition
                  +- *(1) Project [__index_level_0__#0L, A#1, monotonically_increasing_id() AS _w1#38L, B#2, monotonically_increasing_id() AS _w3#39L]
                     +- Scan ExistingRDD[__index_level_0__#0L,A#1,B#2]
```

There will be Project ~ Window ~ Sort sets column number times.

Reply:
@ueshin thanks for the comment. yes i'll refactoring those logics. thanks!

Reply:
This is a very good point .. I wonder if we can work around by defining a temp column and use it.

Reply:
@HyukjinKwon thanks for the advice! let me try :)

Reply:
fixed like below :)

```python
kdf = ks.DataFrame([[2.0, 1.0], [3.0, None], [1.0, 0.0]], columns=list('AB'))
>>> kdf.cummax().explain()
== Physical Plan ==
*(3) Project [__index_level_0__#131L, _we0#153 AS A#169, CASE WHEN isnull(B#133) THEN null ELSE _we1#154 END AS B#170]
+- Window [max(A#132) windowspecdefinition(__natural_order__#139L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS _we0#153, max(B#133) windowspecdefinition(__natural_order__#139L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS _we1#154], [__natural_order__#139L ASC NULLS FIRST]
   +- *(2) Sort [__natural_order__#139L ASC NULLS FIRST], false, 0
      +- Exchange SinglePartition
         +- *(1) Project [__index_level_0__#131L, A#132, __natural_order__#139L, B#133]
            +- *(1) Project [__index_level_0__#131L, A#132, B#133, monotonically_increasing_id() AS __natural_order__#139L]
               +- Scan ExistingRDD[__index_level_0__#131L,A#132,B#133]
```

Reply:
cool! thanks for the fix!

Reply:
Thnaks, My pleasure. :)
BTW, it seems we have some problem in groupby.
Let me check a few hours ago.